test/debug_tools_test.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import subprocess
+
+import pytest
+
+import ray
+
+
+@pytest.fixture(params=[("PLASMA_STORE"), ("RAYLET")])
+def ray_gdb_start(request):
+    # Setup environment and start ray
+    _environ = os.environ.copy()
+    os.environ["RAY_{}_GDB".format(request.param)] = "1"
+    os.environ["RAY_{}_TMUX".format(request.param)] = "1"
+    ray.init(num_cpus=1)
+
+    # Yield expected process keyword
+    if "PLASMA" in request.param:
+        yield "plasma/plasma_store_server"
+    elif "RAYLET" in request.param:
+        yield "raylet/raylet"
+
+    # Restore original environment and stop ray
+    os.environ.clear()
+    os.environ.update(_environ)
```

Comment:
why these two lines instead of just `os.environ = _environ`?

Reply:
nevermind, I think the answer is that `os.environ` is not a dict but rather some special type and the `setattribute` method probably does something special